platform/security/core/security-core-impl/src/main/java/ddf/security/assertion/impl/SecurityAssertionImpl.java

```diff
@@ -337,6 +369,45 @@ public SecurityToken getSecurityToken() {
         return securityToken;
     }
 
+    /**
+     * Checks if the NameIDFormat is of the following formats below, if not, the name is changed
+     * to the value of the first matching usernameAttribute.
+     */
+    public void identifyNameIDFormat() {
+        if ((StringUtils.containsIgnoreCase(nameIDFormat, SAML2Constants.NAMEID_FORMAT_PERSISTENT)
```

Comment:
Just negate the entire boolean in this if and move the code from the else up into this block.